### PR TITLE
Rubocop recently landed modules

### PR DESCRIPTION
**Pull request by adfoster-r7**
Note that the first commit is from https://github.com/rapid7/metasploit-framework/pull/14733 - once that is landed, this will be rebased and the commit will no longer appear in this PR

## Verification

- Spot check the Rubocop autocorrections are valid
- Ensure that CI passes

### CODE REVIEW

File: modules/auxiliary/gather/windows_secrets_dump.rb [hunk 1]
```diff
@@ -68,29 +68,29 @@ class CacheData < BinData::Record
 
     endian :little
 
-    string   :enc_hash, length: 16
```
Comment (adfoster-r7) [inline modules/auxiliary/gather/windows_secrets_dump.rb:71]: Will look into an alternative for bindata formatting

**Comment by gwillcox-r7** [comment]
Looks like this is failing atm on this error: `modules/auxiliary/gather/windows_secrets_dump.rb:1005 - [ERROR] fail_with requires a valid Failure:: reason as first parameter: fail_with( `. Looking at the error codes being used though they all look valid although some are weirdly multiline wrapped so I'm guessing this may be a parsing error.

**Comment by bcoles** [comment]
> 
> 
> Looks like this is failing atm on this error: `modules/auxiliary/gather/windows_secrets_dump.rb:1005 - [ERROR] fail_with requires a valid Failure:: reason as first parameter: fail_with( `. Looking at the error codes being used though they all look valid although some are weirdly multiline wrapped so I'm guessing this may be a parsing error.

At a guess: its due to not using parenthesis `()`.

**Comment by adfoster-r7** [comment]
Thanks for the heads up, will circle back to the `fail_with` error :+1:

Created a separate PR for ignoring Rubocop's ExtraSpacing rules for BinData objects: https://github.com/rapid7/metasploit-framework/pull/14762

**Comment by adfoster-r7** [comment]
I think this is good to ship now. Will hold off on merging until the [zeitwerk rex PR](https://github.com/rapid7/metasploit-framework/pull/14696) to avoid any merge conflicts with that PR

**Pull request merged**